clients/vault/src/oracle/collector/proof_builder.rs

```diff
@@ -158,11 +171,12 @@ impl ScpMessageCollector {
 		match tx_set {
 			Some(res) => Some(res),
 			None => {
+				tracing::info!("get_txset(): FOR SLOT {slot} check_slot_still_recoverable_from_overlay: LAST SLOT INDEX: {}",self.last_slot_index());
 				// If the current slot is still in the range of 'remembered' slots
 				if check_slot_still_recoverable_from_overlay(self.last_slot_index(), slot) {
-					self.fetch_missing_txset_from_overlay(slot, sender).await;
+					self.ask_overlay_for_txset(slot, sender).await;
 				} else {
-					tokio::spawn(self.get_txset_from_horizon_archive(slot));
+					self.get_txset_from_horizon_archive(slot).await;
```

Comment:
So we are now doing `await` on this vs spawning in a new thread as before. This means if I understand correctly that `build_proof` will now wait [here](https://github.com/pendulum-chain/spacewalk/pull/545/files#diff-732d3f237cd7cd9cf03fc64a9ae387ce0e80ad0928e995a3b9e61df4a4c24429R212) and the loop won't continue and call `build_proof` again. Why don't we need this behavior anymore now?

Reply:
Before tokio-console: when the vaults hang, the last logs were in `build_proof` section getting data from the archive. 

I wanted to eliminate unnecessary tasks as possible, especially since these are not monitored at all. Should the original `tokio::spawn(self.get_txset_from_horizon_archive(slot));` become a zombie, there is nothing to stop that task. 

**The time spent looping `build_proof` continuously while the map is empty** 
_VS_
**The time spent waiting for the archive to fill the map** 

is just the same.

Reply:
I see! makes sense, in fact makes much more sense since `get_txset_from_horizon_archive` will return when the `tx_set` was found. But before we were potentially actually spawning many instances of `get_txset_from_horizon_archive` for the same slot, I thought maybe we needed several of these tasks, or that we needed to continue and return a proof with `None` set.